src/components/pages/Home/Onboarding/Main.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { ReactElement, useEffect } from 'react'
+import { animated, useSpringRef, useTransition } from 'react-spring'
+import styles from './Main.module.css'
+
+export default function Main({
+  currentStep,
+  navigationDirection,
+  steps
+}: {
+  currentStep: number
+  navigationDirection: 'prev' | 'next'
+  steps: {
+    shortLabel: string
+    component: ReactElement
+  }[]
+}): ReactElement {
+  const transRef = useSpringRef()
+  const transitions = useTransition(currentStep, {
+    ref: transRef,
+    keys: null,
+    initial: { opacity: 1, transform: 'translate3d(0%,-50%,0)' },
+    from: {
+      opacity: 0,
+      transform: `translate3d(${
+        navigationDirection === 'prev' ? '-100%' : '100%'
+      },-50%,0)`
+    },
+    enter: { opacity: 1, transform: 'translate3d(0%,-50%,0)' },
+    leave: {
+      opacity: 0,
+      transform: `translate3d(${
+        navigationDirection === 'prev' ? '50%' : '-50%'
+      },-50%,0)`
+    },
+    config: { mass: 1, tension: 140, friction: 18 }
+  })
```

Comment:
issue(blocking): readability could be improved here. Maybe it helps if we use values like "left", "right" and so on instead of the pure numbers.